src/mlpack/methods/random_forest/extra_trees_main.cpp

```diff
@@ -0,0 +1,216 @@
+                                  "node.",
+             "n", 1);
+PARAM_INT_IN("maximum_depth", "Maximum depth of the tree (0 means no limit).",
+             "D", 0);
+PARAM_MATRIX_OUT("probabilities", "Predicted class probabilities for each "
+                                  "point in the test set.",
+                 "P");
+PARAM_UROW_OUT("predictions", "Predicted classes for each point in the test "
+                              "set.",
+               "p");
+
+PARAM_DOUBLE_IN("minimum_gain_split", "Minimum gain needed to make a split "
+                                      "when building a tree.",
+                "g", 0);
+PARAM_INT_IN("subspace_dim", "Dimensionality of random subspace to use for "
+                             "each split.  '0' will autoselect the square root of data dimensionality.",
+             "d", 0);
+
+PARAM_INT_IN("seed", "Random seed.  If 0, 'std::time(NULL)' is used.", "s", 0);
+PARAM_FLAG("warm_start", "If true and passed along with `training` and "
+                         "`input_model` then trains more trees on top of existing model.",
+           "w");
+class ExtraTreesModel
+{
+    //The model itself
+    ExtraTrees<> et(data, fullLabels, 3, weights, 20, 1);
+    ExtraTreesModel();
+    //Serialize the model
+    template <typename Archive>
+    void serialize(Archive &ar, const uint32_t /* version */)
+    {
+        ar(CEREAL_NVP(rf));
+    }
+};
+
+PARAM_MODEL_IN(ExtraTreesModel, "input_model", "Pre-trained random forest to "
+                                               "use for classification.",
+               "m");
+PARAM_MODEL_OUT(RandomForestModel, "output_model", "Model to save trained "
+                                                   "random forest to.",
+                "M");
+static void mlpackMain()
+{
+    // Check for incompatible input parameters.
+    if (!IO::HasParam("warn_start"))
+        RequireOnlyOnePassed({"training", "input_model"}, true);
+    // When warn_start is passed, training and input_model must also be passed.
+    RequireNoneOrAllPassed({"warn_start", "training", "input_model"}, true);
+
+    ReportIgnoredParam({{"training", false}}, "print_training_accuracy");
+    ReportIgnoredParam({{"test", false}}, "test_labels");
+
+    RequireAtLeastOnePassed({"test", "output_model", "print_training_accuracy"},
+                            false, "the trained forest model will not be used or saved");
+    if (IO::HasParam("training"))
+    {
+        RequireAtLeastOnePassed({{"labels"}, true, "must pass labels when training set is given"});
+    }
+    if (IO
+        : HasParam("input_model"))
+        model = IO::GetParam << ExtraTreesModel > ("input_model");
+    else
+        model = new ExtraTreesModel();
+    if (IO ::HasParam("training"))
+    {
+        Timer::Start("rf_training");
+        //Training the model on the given input data.
+        mat data = move(IO::.GetParam<mat>("training"));
```

Comment:
```suggestion
        mat data = std::move(IO::GetParam<mat>("training"));
```

Reply:
I had a doubt in this part, if I have already declared using namespace std in the beginning of the program, why do I need to add std here as well?